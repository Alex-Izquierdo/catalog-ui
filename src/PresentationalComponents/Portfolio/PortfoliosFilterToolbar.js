import React, { Component } from 'react';
import { Toolbar } from '@patternfly/react-core';
import OrderToolbarItem from '../Shared/OrderToolbarItem';
import FilterToolbarItem from '../Shared/FilterToolbarItem';
import '../Shared/toolbarschema.scss';

class PortfoliosFilterToolbar extends Component {
  render() {
    return (
      <Toolbar className="searchToolbar">
        <FilterToolbarItem { ...this.props } placeholder={ 'Find a Portfolio' }/>
        <OrderToolbarItem { ...this.props }/>
      </Toolbar>);
  };
};

export default PortfoliosFilterToolbar;
