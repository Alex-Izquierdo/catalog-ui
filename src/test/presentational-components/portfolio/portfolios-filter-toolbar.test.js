import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PortfoliosFilterToolbar from '../../../PresentationalComponents/Portfolio/PortfoliosFilterToolbar';

describe('<PortfoliosFilterToolbar />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      onFilterChange: jest.fn(),
      filterValue: ''
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PortfoliosFilterToolbar { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should call filter action', () => {
    const onFilterChange = jest.fn();
    const wrapper = mount(
      <MemoryRouter>
        <PortfoliosFilterToolbar { ...initialProps } onFilterChange={ onFilterChange } />
      </MemoryRouter>
    );

    const input = wrapper.find('input');
    input.getDOMNode.value = 'foo';
    input.simulate('change');
    expect(onFilterChange).toHaveBeenCalled();
  });
});
