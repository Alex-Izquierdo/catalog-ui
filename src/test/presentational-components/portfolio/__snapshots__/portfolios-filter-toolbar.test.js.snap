// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfoliosFilterToolbar /> should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "componentMapper": Object {
        "Button": [Function],
        "FilterToolbarItem": [Function],
        "Level": [Function],
        "LevelItem": [Function],
        "Link": [Function],
        "Toolbar": [Function],
        "ToolbarGroup": [Function],
        "ToolbarItem": [Function],
        "TopToolbar": [Function],
        "TopToolbarTitle": [Function],
      },
      "render": [Function],
    }
  }
>
  <Component
    component="TopToolbar"
    key="portfolios-top-toolbar"
  >
    <Component
      component="TopToolbarTitle"
      key="portfolios-toolbar-title"
      title="Portfolios"
    />
    <Component
      component="Level"
      key="porftolios-actions"
    >
      <Component
        className="pf-u-mt-md"
        component="Toolbar"
        key="main-portfolio-toolbar"
      >
        <Component
          component="ToolbarGroup"
          key="filter-group/single-toolbar-item-group"
        >
          <Component
            component="ToolbarItem"
            key="filter-group/single-toolbar-item"
          >
            <Component
              component="FilterToolbarItem"
              key="filter-input"
              onFilterChange={[MockFunction]}
              searchValue=""
            />
          </Component>
        </Component>
        <Component
          component="ToolbarGroup"
          key="portfolio-button-group/single-toolbar-item-group"
        >
          <Component
            component="ToolbarItem"
            key="portfolio-button-group/single-toolbar-item"
          >
            <Component
              className=""
              component="Link"
              key="create-portfolio-button/button-link"
              to="/portfolios/add-portfolio"
            >
              <Component
                aria-label="Create portfolio"
                component="Button"
                key="create-portfolio-button"
                title="Create portfolio"
                variant="primary"
              />
            </Component>
          </Component>
        </Component>
      </Component>
      <Component
        component="LevelItem"
        key="pagination-item"
      >
        <Component
          apiRequest={[Function]}
          component={[Function]}
          key="porftolios-pagination"
        />
      </Component>
    </Component>
  </Component>
</ContextProvider>
`;
